safe_transaction_service/history/indexers/tx_processor.py

```diff
@@ -235,9 +237,10 @@ def swap_owner(
         MultisigConfirmation.objects.remove_unused_confirmations(
             contract_address, safe_status.nonce, owner
         )
+        safe_message_models.SafeMessageConfirmation.objects.filter(owner=owner).delete()
```

Comment:
If `SafeMessageConfirmation` have a owner `foreignkey` with delete on cascade shouldn't remove the `messageconfirmation` if the owner is removed?

Reply:
`SafeMessageConfirmation` does not have a foreign key to the owner right? It has to the `SafeMessage`. If the `SafeMessage` is removed then the associated confirmations should be removed too.

Reply:
That's it, no foreign key to the owner